examples/consumergroup/main.go

```diff
@@ -116,7 +116,6 @@ func main() {
 			if ctx.Err() != nil {
 				return
 			}
-			consumer.ready = make(chan bool)
```

Comment:
If we assign a new channel here in goroutine, the `Setup` will only close this new channel and the `<-consumer.ready` on line 123 below will be waiting on the older channel. So even if the new channel is closed, the wait will never end.